### PR TITLE
Add some generated example files to the .gitignore list

### PR DESCRIPTION
**Pull request by yazug**
After running make -k check I was left with a bunch of generated files showing up as untracked files.  Added some rules to .gitignore to deal with them for python and perl5 to reduce how many showed up

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -157,7 +157,36 @@ Examples/test-suite/octave/*.oct
...
+Examples/python/import_packages/same_modnames2/pkg1/pkg2/foo.py
+Examples/python/import_template/bar.py
+Examples/python/import_template/base.py
+Examples/python/import_template/foo.py
+Examples/python/import_template/spam.py
 
 # Scilab generated files
 loader.sce
 
+# Pearl Examples generated files
```
Comment (ojwb) [inline .gitignore]: "Perl"

Comment (yazug) [inline .gitignore]: Good catch, my bad spelling
fixed.

**Comment by yazug** [comment]
I like the clean up you did in addition to this pull request.

**Comment by wsfulton** [comment]
Thanks, the only downside is it requires git 1.8 or later for the ** patterns to work.

**Pull request merged**